### PR TITLE
Changed Git setup to reflect the latest SSH ed25519 recommendations from Github Docs

### PR DESCRIPTION
**Pull request by ManonLef**
- [X] You have read [The Odin Projects Contributing Guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md).
 - [X] The title summarizes the change and where it happened, for example: "Fixes punctuation in Clean Code lesson".
 - [X] If the PR is related to an open issue, use a [relevant keyword](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests) and reference it with the `#` sign and the issue number.
 - [X] You have previewed your markdown (if any) in the [Odin Markdown Preview Tool](https://www.theodinproject.com/lessons/preview)
 - [X] If changes are requested, please make the changes in a timely manner. After you submit the changes, request another review from the maintainer (top right).

#### 1. Describe the changes made and include why they are necessary or important:

Changed the lesson to setup an SSH with an Ed25519 algorithm. This is the latest method in the Github Docs. https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent

In the current version it's RSA focused. It also states that if after the command `ls ~/.ssh/id_rsa.pub` the terminal shows: “No such file or directory”, there's no SSH key present. This is incorrect as it merely means there's no id_rsa.pub, we don't know with that command if the directory is present, and we only searched for an id_rsa.pub file, not the other options like Ecdsa or Ed25519. Changing this to specify exactly what we were looking for, namely an id_ed25519.pub version in this change, makes this part correct. 

Currently there are quite a few help requests on discord from people who state that they already have an SSH key but who still get the  “No such file or directory”. These threads and proposed solutions often end up in generating a new RSA key through TOP instructions while they might already have an Ed25519 installed from different tutorials for example.

Long story short, I think this update is more current and more correct with the Ed25519 focus. Plus it might lead to fewer help requests with the incorrect conslusion from the “No such file or directory” reworded. 

This issue and the proposed changes have been discussed in discord with BriggsE here: https://discord.com/channels/505093832157691914/931168767281610832/931170082674065459

Some decisions were made in the linked Discord thread regarding the choice for Ed25519 specifically and thus diverting a bit from the Github Docs for simplicity's sake.

#### 2. Related Issue
N/A

Closes #XXXXX

### CODE REVIEW

**Review by I3uckwheat — APPROVED** [review]
Looks good to me, thanks for taking this on!

I'll defer to @zachmmeyer and @xandora for feedback though

**Review by zachmmeyer — APPROVED** [review]
Just tested this out and it works for me! Thanks for getting this out!

**Comment by xandora** [comment]
Looks good to me.
